### ISSUE TITLE
Functionality

### ISSUE THREAD

**Issue by JacobJershin** [CSAllenISD/2023-ISP-ModelScheduler#11]
I thought it might be a good idea to open up an issue for functionality. This likely wont be very active until the later phases of the project but I might as well so we can get some discussion going.

Boden wrote a little thing that creates every possible schedule combination pretty much just like Mr. BY wanted. https://www.codermerlin.com/users/bo-hong-chen/Digital%20Portfolio/js/schedule/index.html

**Comment by Muq2** [comment]
What about conflicts?

**Comment by Muq2** [comment]
We should get the basic functions of login, adding/removing courses, etc. before implementing the ability to generate random schedules.

**Comment by Boden-C** [comment]
that was a TEST to see how many schedules would be generated given a master schedule and a selected schedule, basically useless now

it does take into consideration last-last year's master schedule and period availability but not location conflicts. It was more to see if we were given a master schedule and told to make a student's schedule, how much schedules would be generated.

Around 400 schedules were generated but adding a filter it would narrow it down

Since now were creating the master schedule itself it isn't that useful

tldr; it was a test to see schedules generated, this isn't that important, and this should not be in progress right now

**Comment by Muq2** [comment]
> We should focus on an MVP, basic scheduling by drag and drop and printing. We should also keep the idea that analytics and login could be required, so we should focus on making our code expandable. We will discuss the first steps of our project. Starting with zero is setting up Vapor/SQL with a frontend/backend connection using a REST API.

Yes once we get the database design finalized we can create the sql database with the scheme

**Comment by Boden-C** [comment]
OK ive been informed that we actually are given the master schedule, so this might be important again

this is still inspiration for future
